resources/smcp-templates/v2.1/external

```diff
@@ -0,0 +1,107 @@
+apiVersion: maistra.io/v2
+kind: ServiceMeshControlPlane
+metadata:
+  name: external
+spec:
+  version: v2.1
+  gateways:
```

Comment:
Unless you are changing default behavior, the ideal thing to do here would be to reuse the other templates and apply your changes here.  You can refer to the servicemesh and maistra templates for examples.  For example:
```yaml
  profiles:
  - servicemesh
  gateways:
    enabled: false # I don't think anything else is required here, as this disables all gateway charts
  general:
    controlPlaneMode: External
  addons:
    kiali:
      enabled: false
    grafana:
      enabled: false
```

Also, generally speaking, if there are requirements for field settings based on the controlPlaneMode, then I would manage that logic in the render code.  This will provide a single place to maintain derived settings, and you won't have to worry about users overriding them by accident.

Reply:
Thank you, I think I will try to rewrite the template to this format.
In our case the user shouldn't reach the SMCP settings of the control plane side (when controlPlaneMode is enabled). However in the data plane logic we need to consider this